test/unit/picolibrary/indicator/gpio_output_pin_fixed_intensity_indicator/main.cc

```diff
@@ -0,0 +1,176 @@
+/**
+ * picolibrary
+ *
+ * Copyright 2020-2021, Andrew Countryman <apcountryman@gmail.com> and the picolibrary
+ * contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this
+ * file except in compliance with the License. You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+/**
+ * \file
+ * \brief picolibrary::Indicator::GPIO_Output_Pin_Fixed_Intensity_Indicator unit test
+ *        program.
+ */
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+#include "picolibrary/error.h"
+#include "picolibrary/gpio.h"
+#include "picolibrary/indicator.h"
+#include "picolibrary/result.h"
+#include "picolibrary/testing/unit/error.h"
+#include "picolibrary/testing/unit/gpio.h"
+#include "picolibrary/testing/unit/indicator.h"
+#include "picolibrary/testing/unit/random.h"
+#include "picolibrary/void.h"
+
+namespace {
+
+using ::picolibrary::Error_Code;
+using ::picolibrary::Result;
+using ::picolibrary::Void;
+using ::picolibrary::GPIO::Initial_Pin_State;
+using ::picolibrary::Indicator::Initial_Indicator_State;
+using ::picolibrary::Testing::Unit::Mock_Error;
+using ::picolibrary::Testing::Unit::random;
+using ::testing::_;
+using ::testing::Return;
+
+using Indicator =
+    ::picolibrary::Indicator::GPIO_Output_Pin_Fixed_Intensity_Indicator<::picolibrary::Testing::Unit::GPIO::Mock_Output_Pin>;
+
+} // namespace
+
+/**
+ * \brief Verify
+ *        picolibrary::Indicator::GPIO_Output_Pin_Fixed_Intensity_Indicator::initialize()
+ *        properly handles an initialization error.
+ */
+TEST( initialize, initializationError )
+{
+    auto indicator = Indicator{};
+
+    auto const error = random<Mock_Error>();
+
+    EXPECT_CALL( indicator, initialize( _ ) ).WillOnce( Return( error ) );
+
+    auto const result = indicator.initialize( random<Initial_Indicator_State>() );
+
+    EXPECT_TRUE( result.is_error() );
+    EXPECT_EQ( result.error(), error );
+}
+
+/**
+ * \brief Verify
+ *        picolibrary::Indicator::GPIO_Output_Pin_Fixed_Intensity_Indicator::initialize()
+ *        works properly.
+ */
+TEST( initialize, worksProperly )
+{
+    struct {
+        Initial_Indicator_State requested_state;
+        Initial_Pin_State       actual_state;
```

Comment:
While `requested_state` and `actual_state` work well enough for the active low output pin unit tests, I don't know that they work well in this case. Consider renaming these members to something along the lines of the following:

```suggestion
        Initial_Indicator_State indicator_state;
        Initial_Pin_State       pin_state;
```